tianshou/policy/base.py

```diff
@@ -185,6 +185,11 @@ def process_fn(
     def learn(self, batch: Batch, **kwargs: Any) -> Dict[str, Any]:
         """Update policy with a given batch of data.
 
+        If you only want :meth:`learn` calls to accumulate gradients and not
+        actually update the weight, you can specify ``accumulate_grad=True``
+        in the argument. This doesn't block calls to function calls like
+        ``sync_weight``, please use with caution.
```

Comment:
```suggestion
        ``sync_weight``, please use with caution (see issue#437).
```

Reply:
Directly passing a dummy optimizer is an easy way to deal with this, but it will take away the choice of "whether to accumulate grads". Can you elaborate more on this: 

> Like the wrapper approach outside optim.zero_grad()and optim.step()

Reply:
> it will take away the choice of "whether to accumulate grads"

I mean you can set up your choice by passing into different `optim`



> Can you elaborate more on this:

For example, create a new optimizer that inherit existing pytorch optimizer (e.g., Adam):

```python
class MyAwesomeOptim(torch.optim):

  def __init__(self, accumulate_grad = True, *args, **kwargs):
    super().__init__(*args, **kwargs)
    self.opt = accumulate_grad

  def zero_grad(self):
    if self.opt:
        super().zero_grad()

  def step(self):
    if self.opt:
      super().step()
```

Reply:
I agree, the optimizer idea is more concise and looks way better. I'll close this pr then?

Reply:
Sure!